sourcegraph/sourcegraph
# (feat) add ability to allow anonymous usage mode based on "auth.public" site config and "allow-anonymous-usage" license tag

Closes https://github.com/sourcegraph/sourcegraph/issues/52439.

## Test plan
- `sg start dotcom`
- Generate license with `allow-anonymous-usage` tag
- Restart in enterprise mode: `sg start enterprise`
- Set new license
- Set `auth.public=true`
- Check that instance allows searching and browsing public repositories

## Demo

https://github.com/sourcegraph/sourcegraph/assets/6717049/344f9b1e-f54a-4148-9bc8-f0f3d3bb4cd7
